### ISSUE TITLE
`react/destructuring-assignment`: Enforce destructuring props at function parameter

### ISSUE THREAD

**Issue by golopot** [yannickcr/eslint-plugin-react#3232]
I want an option to enforce props destructuring happens in function parameter.

Bad:
```jsx
function Foo(props) {
  const {a} = props
  return <>{a}</>
}
```

Good:
```jsx
function Foo({ a }) {
  return <>{a}</>
}
```

This can be implemented as an option in for `react/destructuring-assignment`. It only affects function components. It does not affect class components.

**Comment by ljharb** [comment]
Seems reasonable, as long as it doesn’t warn when there’s a second reference of the props object.

**Comment by dangerismycat** [comment]
Ah perfect, I've been looking for a rule like this. Has this been released yet? (I think the answer is 'no')

**Comment by ljharb** [comment]
No, it hasn't.
